lib/api/policy.go

```diff
@@ -117,6 +117,10 @@ type PolicySpec struct {
 
 	// PreDNAT indicates to apply the rules in this policy before any DNAT.
 	PreDNAT bool `json:"preDNAT,omitempty"`
+
+	// Types indicates whether this policy applies to ingress, or to egress, or to both.  An
+	// empty or nil value here indicates both ingress and egress.
+	Types []string `json:"types,omitempty" validate:"omitempty,dive,policytype"`
```

Comment:
Might be nice if instead of a string we used a new type for this, so calling code could make use of the constants.  I think this is fairly standard practice for fields like this.

e.g.

```go
type PolicyType string

const (
    PolicyTypeIngress PolicyType = "ingress"
    PolicyTypeEgress PolicyType = "egress"
)
```